spec/design/xql.md

```diff
@@ -0,0 +1,784 @@
+can be understood in isolation ("The SWAPI-name of
+`https://swapi.dev/api/people/1/` is 'Luke Skywalker'"). Results can be
+transported, cached, and merged with other results and always remain a correct
+subset of the information in the original data source.
+
+Let's illustrate with some examples. We'll use the data offered by
+[SWAPI](https://swapi.dev/), the Star Wars API. This API offers a rich set of
+data in a JSON format which is easily read as JSON-LD by attaching a context to
+each resource.
+
+The context:
+
+```json
+{
+  "@context": {
+    "@vocab": "http://swapi.dev/documentation#",
+    "url": "@id",
+    "height": { "@type": "xsd:integer" },
+    "mass": { "@type": "xsd:integer" },
+    "vehicles": { "@type": "@id" }
+  }
+}
+```
+
+Additionally, we add an additional `@type` key to each resource naming its type,
+as this is not present in SWAPI's responses and [not possible to add with a
+`@context`](https://github.com/w3c/json-ld-syntax/issues/386). The values for
+`@type` are `Planet`, `Spaceship`, `Vehicle`, `Person`, `Film`, and `Species`.
+
+#### Match by `@id`: What color is Luke's hair?
+
+Suppose we have the IRI for [Luke Skywalker](https://swapi.dev/api/people/1/). Then we might ask:
+
+```json
+{
+  "@context": { "@vocab": "http://swapi.dev/documentation#" },
+  "@id": "https://swapi.dev/api/people/1/",
+  "hair_color": "?"
+}
+```
+
+Note the placeholder at `hair_color`. The query engine will attempt to find a
+value for this key. We will get the result:
+
+```json
+{
+  "@context": { "@vocab": "http://swapi.dev/documentation#" },
+  "@id": "https://swapi.dev/api/people/1/",
+  "hair_color": "blond"
+}
+```
+
+As JSON-LD, this describes exactly one fact, "Luke's hair is blond", written in Turtle as:
+
+```turtle
+<https://swapi.dev/api/people/1/> <http://swapi.dev/documentation#hair_color> "blond" .
+```
+
+This is a (very small) subgraph of the original graph: a single triple which was
+present in the original data.
+
+#### Match by `name`: What color are Luke's eyes?
+
+Suppose we _don't_ know Luke's IRI yet. Then we might ask:
+
+```json
+{
+  "@context": { "@vocab": "http://swapi.dev/documentation#" },
+  "name": "Luke Skywalker",
+  "eye_color": "?"
+}
+```
+
+Then we expect to get the following:
+
+```json
+{
+  "@context": { "@vocab": "http://swapi.dev/documentation#" },
+  "name": "Luke Skywalker",
+  "eye_color": "blue"
+}
+```
+
+Notice that the result, once again, has the same shape as the query. The query
+serves as a template which the query engine "fills in". Notice also that what
+this means in JSON-LD, "Something named 'Luke Skywalker' has blue eyes," is
+again a subgraph of the original data:
+
+```turtle
+[] <http://swapi.dev/documentation#name> "Luke Skywalker" ;
+   <http://swapi.dev/documentation#eye_color> "blue" .
+```
+
+(Techincally speaking, it's more proper to say that the original graph
+[_entails_](https://w3c.github.io/rdf-semantics/spec/#simpleentailment) this
```

Comment:
In the query result; rather than using a blank node.

Is there a counter-example in which multiple characters have blue eyes and the `eye_color` becomes multi-valued, or do you get multiple result subjects?